apps/frontend/src/app/Solver/GOSolver/polyUB.ts

```diff
@@ -0,0 +1,196 @@
+import { customMapFormula, forEachNodes } from "../../Formula/internal";
+import { OptNode, allOperations } from "../../Formula/optimization";
+import { ConstantNode } from "../../Formula/type";
+import { prod, threshold } from "../../Formula/utils";
+import { assertUnreachable, cartesian } from "../../Util/Util";
+import { ArtifactsBySlot, MinMax, computeFullArtRange, computeNodeRange } from "../common";
+import { Linear } from "./BNBSplitWorker";
+
+export type Polynomial = PolyProd | PolySum | LinTerm
+type LinTerm = { type: 'lin', lin: Linear }
+type PolyProd = { type: 'prod', terms: Polynomial[], $k: number }
```

Comment:
Can we switch to `$c` so that they are consistent, or is there a reason for the difference?